bank-transaction-categorizer/src/Handler/CategorizeTransactions.hs

```diff
@@ -24,12 +31,34 @@ instance ToJSON CategorizeTransactionsResult
 
 postCategorizeTransactionsR :: Handler Value
 postCategorizeTransactionsR = do
-    bytes <- rawRequestBody C.$$ CL.fold BS.append BS.empty
+    csvBS <- rawRequestBody C.$$ CL.fold BS.append BS.empty
+    let bankType = detectBankType csvBS
+    let abc = case bankType of
+            ChaseBank -> Csv.decodeByName $ LBS.fromStrict csvBS
+            WellsFargoBank -> Csv.decode Csv.NoHeader $ LBS.fromStrict csvBS
+            UnknownBank -> notFound
     _ <- Import.traceM "Test"
-    _ <- Import.traceM $ show $ BS8.takeWhile (/= '\n') bytes
-    _ <- Import.traceM $ show $ isChaseHeader $ show $ BS8.takeWhile (/= '\n') bytes
-    _ <- Import.traceM $ show $ detectBankType bytes
+    _ <- Import.traceM $ show $ BS8.takeWhile (/= '\n') csvBS
+    _ <- Import.traceM $ show $ isChaseHeader $ show $ BS8.takeWhile (/= '\n') csvBS
+    _ <- Import.traceM $ show $ detectBankType csvBS
 
-    pure $ toJSON result 
+    returnJson result 
     where
-        result = CategorizeTransactionsResult { categorizedTransactions = Map.fromList [("key1", "Food"), ("key2", "Groceries")] }
\ No newline at end of file
+        result = CategorizeTransactionsResult { categorizedTransactions = Map.fromList [("key1", "Food"), ("key2", "Groceries")] }
+
+categorizeChaseTransactions :: Vector Chase.ChaseTransaction -> Vector Transaction
+categorizeChaseTransactions transactions = do
+    -- must parse this into a maybe
+    categoryMapStr <- categorizeTransactions $ toList (Chase.description <$> transactions)
```

Comment:
this is an IO operation and the parent function is pure, so this cant work. however, what is the ideal way to handle this?